routes/api/auth.js

```diff
@@ -1,10 +1,87 @@
 const express = require('express');
 const router = express.Router();
+const bcrypt = require('bcryptjs');
+const auth = require('../../middleware/auth');
+const User = require('../../models/User');
+const jwt = require('jsonwebtoken');
+const config = require('config');
+const { check, validationResult } = require('express-validator');
+
 
 //@route Get api/auth
 //@desc Test route
 //@access Public
 
-router.get('/', (req,res) => res.send('Auth Route'));
+router.get('/', auth, async(req,res) => {
+    try{
+        const user = await User.findById(req.user.id).select('-password');
+        res.json(user);
+    }catch(err){
+        console.error(err.message);
+        res.status(500).send('Server Error');
+    }
+});
+
+//@route Post api/users
+//@desc Register user
+//@access Public
+
+router.post('/', 
+[
+    
+    check('email', 'Email is required/should be valid').isEmail(),
+    check('password', 'Passwored Required').exists()
+],
+
+async (req,res) => {
+    
+    const errors = validationResult(req);
+    if(!errors.isEmpty()){
+        return res.status(400).json({ errors: errors.array() });
+    }
+
+    const { email, password } = req.body;
+
+    try{
+        let user = await User.findOne({ email });
+        
+        //See if user exists
+
+         if(!user){
+           return res.status(400).json({errors: [{msg: 'Invalid Credentials'}]});
+        }
+        
+        const isMatch = await bcrypt.compare(password, user.password);
```

Comment:
Why do we need to use await here?
where is password and user.password coming from?

Reply:
Changes looks good but just if you can mention that do we really need await here?